insonmnia/locator/util.go

```diff
@@ -0,0 +1,26 @@
+package locator
+
+import (
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/net/context"
+
+	"github.com/sonm-io/core/util"
+)
+
+func ExtractEthAddr(ctx context.Context) (common.Address, error) {
+	pr, ok := peer.FromContext(ctx)
+	if !ok {
+		return common.Address{}, status.Error(codes.DataLoss, "failed to get peer from ctx")
```

Comment:
DataLoss?

Reply:
@noxiouz Could help answer to this question, I guess.

Reply:
Thank you very much for taking your time reviewing this PR. :)

Reply:
my bad. Probably, UNAUTHENTICATED is better